### PR TITLE
(FM-2752) Add/update travis with modulesync

### PR DESCRIPTION
**Pull request by cyberious**

### CODE REVIEW

File: spec/spec.opts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+--format
+s
+--colour
+--loadby
+mtime
+--backtrace
```
Comment (ferventcoder) [inline spec/spec.opts]: As long as all of this works when I run specs from windows.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+clone_depth: 10
+init:
+- SET
+- 'mkdir C:\ProgramData\PuppetLabs\code && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\facter && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\hiera && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\puppet\var && exit 0'
+environment:
+  matrix:
+  - PUPPET_GEM_VERSION: 3.7.5
```
Comment (ferventcoder) [inline appveyor.yml]: You need https://github.com/puppetlabs/puppetlabs-reboot/blob/master/metadata.json#L92 3.3.0, then 3.8.1 and 4.0.0

File: spec/spec.opts (hunk 1 above)
Comment (cyberious) [inline spec/spec.opts]: Have not seen issues with appveyor for wsus_client doing this

File: appveyor.yml (hunk 2 above)
Comment (cyberious) [inline appveyor.yml]: In the long term perhaps, but this is an initial spike for current development, I would like to get our tests working in ~>3.7 and ~>4.0 first than we can take on the load to get them to work back to 3.3

File: .travis.yml [hunk 3]
```diff
@@ -1,30 +1,21 @@
...
-      env: PUPPET_GEM_VERSION="~> 3.1.0"
-    - rvm: 1.8.7
-      env: PUPPET_GEM_VERSION="~> 3.2.0"
+  include:
+  - rvm: 1.9.3
+    env: PUPPET_GEM_VERSION="~> 3.0"
+  - rvm: 2.1.5
+    env: PUPPET_GEM_VERSION="~> 3.0"
+  - rvm: 2.1.5
+    env: PUPPET_GEM_VERSIONS="~> 4.0"
```
Comment (Iristyle) [inline .travis.yml]: extra `s`

**Comment by Iristyle** [comment]
Looks like we have quite a few test failures....

**Pull request merged**